web/annotation_layer_builder.js

```diff
@@ -119,6 +119,24 @@ class AnnotationLayerBuilder {
       return;
     }
 
+    const uniqueLinks = linkAnnotations.filter(link => {
+      for (const annotation of annotations) {
+        const area = rect =>
+          Math.abs(rect[2] - rect[0]) * Math.abs(rect[3] - rect[1]);
+        const intersect = Util.intersect(annotation.rect, link.rect); // Find the intersection between the annotation and the link.
+        if (
+          annotation.subtype === "Link" &&
```

Comment:
This should compare against the `annotationType`, rather than using string comparison.